rasa_utils/bot_server_channel.py

```diff
@@ -49,7 +53,11 @@ def clear(self, cid):
         self.save()
 
     def save(self):
-        json.dump(self._store, open(self._filename, "w"))
+        try:
+            with open(self._filename, "w") as message_store_file:
+                json.dump(self._store, message_store_file)
+        except Exception as e:
```

Comment:
```suggestion
        except IOError as e:
```

Reply:
Hi @jstriebel , 
* No worries
* I had a look at the methods that were used by the class derived by InputChannel (i.e. BotServerInputChannel), and they were not defined neither in BotServerInputChannel nor in InputChannel, but they were defined in RestInput. One of the two following things might have happened : 
  - Either the programmer who wrote it wanted to define it by subclassing from RestInput and subclassed from InputChannel because of whatever accident, or
  - Originally those methods were in InputChannel from Rasa, but Rasa created a new subclass and moved the methods there. 

In any case BotServerInputChannel should have those methods defined somewhere, my suggestion was to just subclass from RestInput instead of from InputChannel. 

* I added the line "from rasa_core import utils" because "utils" is used explicitly in the code. I'm assuming you did a search for "rasa_core.utils" but could not find it... maybe my explanation was misleading! I added it in the commit in line 18.